sql/expression/function/aggregation/window_framer.go

```diff
@@ -376,19 +412,53 @@ type rangeFramerBase struct {
 	startNFollowing sql.Expression
 	endNPreceding   sql.Expression
 	endNFollowing   sql.Expression
+
+	startInclusion sql.Expression
+	endInclusion   sql.Expression
 }
 
 var _ sql.WindowFramer = (*rangeFramerBase)(nil)
 
-func (f *rangeFramerBase) NewFramer(interval sql.WindowInterval) sql.WindowFramer {
+func (f *rangeFramerBase) NewFramer(interval sql.WindowInterval) (sql.WindowFramer, error) {
+	var startInclusion sql.Expression
+	switch {
+	case f.startCurrentRow:
+		startInclusion = f.expr
+	case f.startNPreceding != nil:
+		startInclusion = expression.NewArithmetic(f.expr, f.startNPreceding, ast.MinusStr)
+	case f.startNFollowing != nil:
+		startInclusion = expression.NewArithmetic(f.expr, f.startNFollowing, ast.PlusStr)
+	}
+
+	// TODO: how to validate datetime, interval pair when they aren't type comparable
+	//if startInclusion != nil && startInclusion.Type().Promote()  != f.expr.Type().Promote() {
+	//	return nil, ErrRangeIntervalTypeMismatch
+	//}
```

Comment:
i tried doing `Type().Promote` comparisons but that doesn't work for dates and intervals. Pushing this into the analyzer seems ideal